core/rawdb/accessors_chain.go

```diff
@@ -373,3 +373,28 @@ func FindCommonAncestor(db DatabaseReader, a, b *types.Header) *types.Header {
 	}
 	return a
 }
+
+// ReadShardState retrieves sharding state
+func ReadShardState(db DatabaseReader, hash common.Hash, number uint64) types.ShardState {
+	data, _ := db.Get(shardStateKey(number, hash))
+	if len(data) == 0 {
```

Comment:
do you need any log here?

Reply:
since not all blocks contains the shard state value, in this case, we just ignore it.